### ISSUE TITLE
Consider maximum recursion depth

### ISSUE THREAD

**Issue by azaika** [WICG/service-worker-static-routing-api#6]
By the proposed Conditional Syntax [[Design Doc]](https://docs.google.com/document/d/1B2D8w_2M4j9CZC1VccBOAhfCdiRP3yPCSV5UKKDhU70), conditions could be recursive (i.e. nested). For example, we can specify `and` conditions as elements of an `or` condition.
However, since allowing infinite nests may cause some problems in implementations, each browser will restrict the depth of nests and it will involve inconsistent behaviors between browsers.
I suggest having max recursion limits in the specification to avoid such confusion.

**Comment by yoshisatoyanagisawa** [comment]
I agree to have maximum recursion limit.  Do we also have a limit to conditions listed in `or` and `and`?  I mean that `or` and `and` can take a list, and writing too long list can also be harmful.

**Comment by yoshisatoyanagisawa** [comment]
https://github.com/w3c/ServiceWorker/pull/1752 defines the max depth.  TypeError will be raised on exceeding the limit.
